core/src/main/java/org/neo4j/graphalgo/core/utils/ConcurrencyConfig.java

```diff
@@ -0,0 +1,33 @@
+package org.neo4j.graphalgo.core.utils;
+
+final class ConcurrencyConfig {
+
+    private static final String PROCESSORS_OVERRIDE_PROPERTY = "neo4j.graphalgo.processors";
+    private static final int MAX_CE_CONCURRENCY = 4;
+
+    final int maxConcurrency;
+    final int defaultConcurrency;
+
+    static ConcurrencyConfig of() {
+        Integer definedProcessors = null;
+        try {
+            definedProcessors = Integer.getInteger(PROCESSORS_OVERRIDE_PROPERTY);
+        } catch (SecurityException ignored) {
+        }
+        if (definedProcessors == null) {
+            definedProcessors = Runtime.getRuntime().availableProcessors();
+        }
+        boolean isOnEnterprise = Package.getPackage("org.neo4j.kernel.impl.enterprise") != null;
```

Comment:
Yes, you're right. I did discuss this with @jexp and we said that this is ok for now because 1) it's just temporary until we got proper scheduling in 4.0 and 2) if you're bothered to build an extra plugin with this package you might as well just remove this check and rebuild the algos plugin and get the same effect.